lua/cfc_entity_stubber/main_loader.lua

```diff
@@ -0,0 +1,79 @@
+cfcEntityStubber = { }
+cfcEntityStubber.stubQueue = { }
+cfcEntityStubber.stubbers = { }
+cfcEntityStubber.oldWeaponStats = { }
+cfcEntityStubber.newWeaponStats = { }
+local stubbersDirectory = "cfc_entity_stubber/stubbers/"
+
+-- MAIN STUBBER FUNCTIONS 
+function cfcEntityStubber.includeStubbers( )
+    local files = file.Find( stubbersDirectory .. "*.lua", "LUA", "datedesc" )
+
+    for _, stubber in ipairs( files ) do
+        MsgC( Color( 41, 41, 41 ), "[", Color( 150, 150, 150 ), "Stubber", Color( 41, 41, 41 ), "] ", Color( 0, 255, 0 ), stubber .. " successfully loaded.\n" )
+        include( stubbersDirectory .. stubber )
+    end
+end
+
+function cfcEntityStubber.getStubs( tab )
+    for _, dir in ipairs( tab ) do
+        local stubFolderPath = "cfc_entity_stubber/" .. dir .. "/"
+        -- List all stubs in the packs folder, sorted by names ascending
+        local stubFiles, stubFolders = file.Find( stubFolderPath .. "*", "LUA", "nameasc" )
+        local emptyFolder = table.IsEmpty( stubFiles )
+        local emptySubFolder = table.IsEmpty( stubFolders )
+
+        if emptyFolder and emptySubFolder then
+            MsgC( Color( 41, 41, 41 ), "[", Color( 150, 150, 150 ), "Stubber", Color( 41, 41, 41 ), "] ", Color( 255, 0, 0 ), "folder " .. dir .. " is empty or doesn't exist.\n" )
+
+            return
+        end
+
+        if not emptySubFolder then
+            for _, stubFile in ipairs( stubFolders ) do
+                local subfolderPath = stubFolderPath .. stubFile
+                local subfolderFiles = file.Find( subfolderPath .. "/*.lua", "LUA", "nameasc" )
+
+                for _, fileName in pairs( subfolderFiles ) do
+                    include( subfolderPath .. "/" .. fileName )
+                end
+            end
+        end
+
+        if not emptyFolder then
+            for _, stubFile in ipairs( stubFiles ) do
+                print( stubFile )
+            end
+        end
+    end
+end
+
+function cfcEntityStubber.runStubs( stubQueue )
+    for _, stub in pairs( stubQueue ) do
+        stub( )
+    end
+end
+
+-- STUBBER FUNCTIONS
+-- Accepts the folder name of the stubbers files
+function cfcEntityStubber.registerStubber( stubberName )
+    table.insert( cfcEntityStubber.stubbers, stubberName )
+end
+
+function cfcEntityStubber.registerStub( stub )
+    table.insert( cfcEntityStubber.stubQueue, stub )
+end
+
+function cfcEntityStubber.getWeapon( wepClass )
+    weapon = weapons.GetStored( wepClass )
+    cfcEntityStubber.newWeaponStats.wepClass = weapon
+
+    return weapon
+end
+
+-- HOOKS
+hook.Add( "InitPostEntity", "StubberStart", function( )
+    cfcEntityStubber.includeStubbers( )
+    cfcEntityStubber.getStubs( cfcEntityStubber.stubbers )
```

Comment:
this ones kidn of nitpicky. but maybe instead of getStubs. something like loadStubs might be a better name

Reply:
aww thanks, youre so nice